paddle/phi/api/yaml/sparse_ops.yaml

```diff
@@ -457,3 +457,15 @@
            mv_csr{sparse_csr, dense -> dense}
     layout : x
   backward: mv_grad
+
+- op : transpose
+  args : (Tensor x, int[] perm)
+  output : Tensor(out)
+  infer_meta :
+    func : TransposeInferMeta
```

Comment:
I think the used " func : TransposeInferMeta"  is actually the one for dense tensor. But the TransposeInferMeta for dense tensor is also applicable to sparse tensor. So I submit a PR to delete maybe unused code in paddle\phi\infermeta\sparse\unary.h 

https://github.com/PaddlePaddle/Paddle/pull/46844